controllers/storagecluster/odfinfoconfig_test.go

```diff
@@ -0,0 +1,144 @@
+package storagecluster
+
+import (
+	"github.com/red-hat-storage/ocs-operator/api/v4/v1alpha1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+	"k8s.io/apimachinery/pkg/types"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+	"testing"
+
+	api "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"gotest.tools/v3/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestOdfInfoConfig(t *testing.T) {
+
+	const namespace = "storage-test-ns"
+	os.Setenv(util.OperatorNamespaceEnvVar, namespace)
+
+	testTable := []struct {
+		label                        string
+		expectedPublicNetworkValue   string
+		storageSystemOwnerRef        metav1.OwnerReference
+		ocsVersion                   string
+		deploymentType               string
+		storageClusterNamespacedName types.NamespacedName
+		cephClusterFSID              string
+		numConnectedClients          int
+		shouldFail                   bool
+	}{
+		{
+			label: "Case #1: Green path no StorageConsumers",
+			storageSystemOwnerRef: metav1.OwnerReference{
+				Name:       "storage-test",
+				Kind:       "StorageSystem",
+				APIVersion: "v1",
+			},
+			ocsVersion:                   "3.16.0",
```

Comment:
@raaizik shouldn't this be `4.16.0`?
And, with the CSV version changed to `4.17` now, it would be `4.17.0`, isn't it?

Reply:
I have opened a PR to fix the failing test case, because of CSV version change. But, if this change is unintentional I can revert it in my open PR.

Reply:
Also, I don't think we should have a dependency of version check like this in the tests, this would mean that we would need to update the version in the test after each release, which is an added task.

Reply:
This test case isn't failing, @Nikhil-Ladha. It's just a dummy for unit test purposes, not sure why you think it should store an up-to-date value.

Reply:
The `ocsversion` is fetched from the storagecluster CR and compared to the one hardcoded in the test case.
And, the storagecluster version is updated in each release.
Also, the test is failing on master branch now, because the CSV is updated to `4.17.0` so, the assertion is failing at `3.17.0 != 3.16.0`.
Ex: https://github.com/red-hat-storage/ocs-operator/actions/runs/9147599372/job/25149216914

Reply:
How come I didn't catch this assertion error from this PR's unit tests check?

Reply:
Why was this change done here: https://github.com/red-hat-storage/ocs-operator/pull/2440/files#diff-694a65a3896ce9e2d76b39af6d0e75202cfda7b30f1f9ea62b65ecb4ab7b44f7R358 ?
It sets the version as `1 less than the actual version` for all the storageclusters created for the unit-test.

Reply:
> How come I didn't catch this assertion error from this PR's unit tests check?

The version update PR was merged before your PR was merged, and you might have missed to rebase the PR on top of latest changes.

Reply:
> Why was this change done here: https://github.com/red-hat-storage/ocs-operator/pull/2440/files#diff-694a65a3896ce9e2d76b39af6d0e75202cfda7b30f1f9ea62b65ecb4ab7b44f7R358 ? It sets the version as `1 less than the actual version` for all the storageclusters created for the unit-test.

The CSV (and OCS version) is required for the new CM which is watched by StorageCluster.

Reply:
That's fine.
What I meant was, why are we setting it to `1 less than the current version`? We should simply set it to current version, isn't it?
Shouldn't we simply be doing:
```
	csv := &opv1a1.ClusterServiceVersion{
		ObjectMeta: metav1.ObjectMeta{
			Name:      fmt.Sprintf("ocs-operator-%s", sc.Name),
			Namespace: sc.Namespace,
		},
		Spec: opv1a1.ClusterServiceVersionSpec{
			Version: oprverion.OperatorVersion{Version: ocsversion.Version},
		},
	}
```
i.e, use the `ocsversion.Version` directly in the `csv` var.

Reply:
Oh, IC. Yep, makes sense to me.

Reply:
Cool, though I was thinking of updating the fixed version string to dynamic version i.e, set it as `version.Version` directly. This would automatically take in the version value of the release.

Reply:
Right

Reply:
@raaizik @Nikhil-Ladha In a unit test you should assume the desired value, not read it from program logic. If you do so you will get into a situation where the test does not test anything as a bug occurs in the underlying logic. 

For example, assume a bug in `version.Version`, if both the Program code and the test read this value then even if we have a bug in the code that fills in this value we will still get the test passing.

Reply:
Well, that is something that should be caught while reviewing the PR.
The current code creates a new dependency of updating the hardcoded value in the test for each release, which is not something we would want to do I assume?
cc @iamniting

Reply:
I think @nb-ohad's suggestion is to resolve that so called dependency by coupling the dummy data with some hard-coded value check.

Reply:
I would not prefer to have hardcoded values which we will need to change every release. I agree with @Nikhil-Ladha here.